engine/lite/components/kafka/src/test/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaAvroFunctionalTest.scala

```diff
@@ -0,0 +1,50 @@
+package pl.touk.nussknacker.engine.lite.components
+
+import org.apache.avro.generic.GenericRecord
+import org.scalatest.{FunSuite, Matchers}
+import pl.touk.nussknacker.engine.avro.decode.BestEffortAvroDecoder
+import pl.touk.nussknacker.engine.avro.encode.ValidationMode
+import pl.touk.nussknacker.engine.avro.schemaregistry.SchemaVersionOption
+import pl.touk.nussknacker.engine.build.ScenarioBuilder
+import pl.touk.nussknacker.engine.lite.util.test.{KafkaConsumerRecord, KafkaTestScenarioRunner, LiteKafkaAvroTestScenarioRunner}
+
+class LiteKafkaAvroFunctionalTest extends FunSuite with Matchers {
+
+  import KafkaTestScenarioRunner._
+  import pl.touk.nussknacker.engine.avro.KafkaAvroBaseComponentTransformer._
+  import pl.touk.nussknacker.engine.spel.Implicits._
+
+  private val RecordSchemaString: String =
+    """{
+      |  "type": "record",
+      |  "namespace": "pl.touk.nussknacker.engine.avro",
+      |  "name": "FullName",
+      |  "fields": [
+      |    { "name": "first", "type": "string" },
+      |    { "name": "last", "type": "string" },
+      |    { "name": "age", "type": "int" }
+      |  ]
+      |}
+    """.stripMargin
+
+  private val inputTopic = "input"
+  private val outputTopic = "output"
+
+  test("should test end to end kafka avro sink / source") {
+    //Given
+    val scenario = ScenarioBuilder.streamingLite("check avro serialization")
+      .source("my-source", SourceName, TopicParamName -> s"'$inputTopic'", SchemaVersionParamName -> s"'${SchemaVersionOption.LatestOptionName}'")
+      .emptySink("my-sink", SinkName, TopicParamName -> s"'$outputTopic'",  SchemaVersionParamName -> s"'${SchemaVersionOption.LatestOptionName}'", SinkKeyParamName -> "", SinkValueParamName -> s"#input", SinkValidationModeParameterName -> s"'${ValidationMode.strict.name}'")
+
+    val runtime = LiteKafkaAvroTestScenarioRunner(new LiteKafkaSourceImplFactory, LiteKafkaAvroSinkImplFactory)
```

Comment:
We someone who want to test his/her kafka avro scenario have to provide this KafkaSourceImpl and KafkaAvroSinkImpl? Why not to make it default?

Reply:
Because of dependencies.. toolkit doesn't have access to Implementation. If add at toolkit dependency to light implementation then we can't do functional tests ad light implementation because of looping dependencies?

Reply:
I see now, It looks like we need to have another module kafka-functional-tests depending on kafka-testkit and than kafka-testkit might depend on on kafka-components.
After that change people could write kafka functional tests without troubles. Can you add at least TODO that this API should be simplified?